src/wallet/mod.rs

```diff
@@ -162,21 +162,87 @@ where
     }
 }
 
+/// The address index selection strategy to use to derived an address from the wallet's external
+/// descriptor. See [`Wallet::get_address`].
+#[derive(Debug)]
+pub enum AddressIndex {
```

Comment:
From the meeting discussion I was expecting 4 elements in the enum, what's missing is the option to return the address at the given index and update the internal index (with all the necessary warning in docs if a user set an index back where they are already used or too far exceeding the gap limit and causing issue with sync)

Reply:
Yes I should have mentioned I didn't add it because it wasn't needed for futurepaul's use-case, but I guess it could still be useful for someone so I'll add it.  I'm going to call it `WalletIndex::Reset(u32)` unless someone has a better name. :slightly_smiling_face:

Reply:
LGTM, the docs may warn also in case of setting the index too forward requires syncing with a non-default gap limit (which is 20)

Reply:
I added this to the `Reset(u32)` docs, does it sound ok? 

"Also if the index is reset to a value earlier than the [`crate::blockchain::Blockchain`] stop_gap (default is 20) then a larger stop_gap should be used to monitor for all possibly used addresses."